docs/introduction/order-processing.md

```diff
@@ -0,0 +1,185 @@
+
+## Create order programmatically
+
+Let's assume we have a product, transport, and payment method that we want to use to create an order programmatically.
+
+```php
+$product = $this->productRepository->getById(1);
+$transport = $this->transportRepository->getById(3);
+$payment = $this->paymentRepository->getById(1);
+```
+
+First, we need to create an instance of the `OrderInput` object, which encapsulates the order input data and populate the product, payment, and transport.
+
+```php
+// orderInputFactory is an instance of \Shopsys\FrameworkBundle\Model\Order\Processing\OrderInputFactory
+$orderInput = $this->orderInputFactory->create($this->domain->getDomainConfigById($domainId));
+
+$orderInput->addProduct($product, 1); // 1 is the desired quantity
+$orderInput->setTransport($transport);
+$orderInput->setPayment($payment);
+```
+
+It is possible to set additional data (without a specific method) to the `OrderInput` object using the `addAdditionalData()` method.
+Usually this data is used by custom middlewares.
+
+```php
+$orderInput->addAdditionalData(MyCustomMiddleware::DATA_KEY, $value);
+```
+
+The data may be retrieved in the middleware from the `OrderInput` object using the `getAdditionalData()` method.
+
+```php
+$value = $orderProcessingData->orderInput->findAdditionalData(MyCustomMiddleware::DATA_KEY);
+```
+
+Next, we create an instance of the `OrderData` object, with basic information.
+These values are usually provided by the customer and may be further used/changed by the middlewares.
+
+```php
+$orderData = $this->orderDataFactory->create();
+$orderData->firstName = 'firstName';
+$orderData->lastName = 'lastName';
+$orderData->email = 'email@example.com';
+$orderData->telephone = '123456879';
+$orderData->street = 'street';
+$orderData->city = 'city';
+$orderData->postcode = '12345';
+$orderData->country = $this->countryFacade->findByCode('CZ');
+$orderData->currency = $this->currencyFacade->getDomainDefaultCurrencyByDomainId($domainId);
+```
+
+Now we are ready to process the order using the `OrderProcessor` service (instance of the`\Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessor` class).
+
+```php
+$orderData = $this->orderProcessor->process(
+    $orderInput,
+    $orderData,
+);
+```
+
+After processing, the order data is ready to be used to create an order.
+
+```php
+// placeOrderFacade is an instance of \Shopsys\FrameworkBundle\Model\Order\PlaceOrderFacade class
+$this->placeOrderFacade->placeOrder($orderData);
+```
+
+!!! note
+
+    You may also call the `PlaceOrderFacade::createOrder` method.<br>
+    This method bypass some steps during the normal place order process, like sending emails, dispatching amqp messages, etc.<br>
+    It is useful when you need to create an order without side effects - for example in tests, during imports, etc.
+
+## Conclusion
```

Comment:
I would add a link to the cookbook `Adding a New Order Item Type` here (or/and anywhere else in the article). Also the cookbook should contain a link to this article

Reply:
added